### PR TITLE
feat: create the cucumber tests

### PR DESCRIPTION
**Pull request by viniciusguardieirozup**
### Related Issues
#879 

### Description and Example

Create a scenario using cucumber

### Checklist

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/6d7444ce71bbfe63f95a41a1fb914d09e63f8720/android/sample/src/main/AndroidManifest.xml#L23">android/sample/src/main/AndroidManifest.xml#L23</a> - On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute <code>android:fullBackupContent</code> to specify an <code>@xml</code> resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/6d7444ce71bbfe63f95a41a1fb914d09e63f8720/android/sample/src/main/AndroidManifest.xml#L31">android/sample/src/main/AndroidManifest.xml#L31</a> - Attribute <code>usesCleartextTraffic</code> is only used in API level 23 and higher (current min is 19)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/sample/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -16,10 +16,12 @@
...
 
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="br.com.zup.beagle.sample">
+    package="br.com.zup.beagle.sample"
+    android:sharedUserId="br.com.zup.beagle.sample.uid">
 
     <uses-permission android:name="android.permission.INTERNET" />
 
+
     <application
```
Comment (botbeaglezup) [inline android/sample/src/main/AndroidManifest.xml:25]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/084740bc7ab792a3b52f1ddb24e3287aba5a1757/android/sample/src/main/AndroidManifest.xml#L25"></span>On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute <code>android:fullBackupContent</code> to specify an <code>@xml</code> resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html</td>
    </tr>
  </tbody>
</table>

File: android/buildSrc/src/main/kotlin/br/com/zup/beagle/Dependencies.kt [hunk 2]
```diff
@@ -94,6 +94,9 @@ class Dependencies : Plugin<Project> {
         const val archCoreTesting = "1.1.0"
         const val espressoCore = "3.0.1"
         const val multidex = "1.0.3"
+        const val cucumberAndroidVersion = "1.2.5@jar"
+        const val cucumberPicocontainer = "1.2.4"
+
```
Comment (diogooliveira-zup) [inline android/buildSrc/src/main/kotlin/br/com/zup/beagle/Dependencies.kt:99]: Hi! Please remove the unecessary break line

File: android/sample/build.gradle [hunk 3]
```diff
@@ -34,7 +34,25 @@ android {
...
+           testInstrumentationRunner "br.com.zup.beagle.sample.SampleTestRunner"
+        } else {
+            testApplicationId "br.com.zup.beagle.sample.cucumber.steps"
+            testInstrumentationRunner "br.com.zup.beagle.sample.cucumber.steps.Instrumentation"
+        }
+
+
+        buildConfigField "String", "TEST_TAGS", "\"${getTestTags()}\""
+
+    }
```
Comment (diogooliveira-zup) [inline android/sample/build.gradle:50]: Same here
```suggestion
        }
        
        buildConfigField "String", "TEST_TAGS", "\"${getTestTags()}\""
    }
```

File: android/sample/build.gradle [hunk 4]
```diff
@@ -70,4 +88,22 @@ dependencies {
...
+    androidTestImplementation Dependencies.TestLibraries.cucumberAndroid
+    androidTestImplementation Dependencies.TestLibraries.cucumberPicocontainer
+}
+
+def getTestTags() {
+    return project.hasProperty("tags") ? project.getProperties().get("tags") : ""
+}
+
+
+screenshots {
```
Comment (diogooliveira-zup) [inline android/sample/build.gradle:101]: same here 
```suggestion
}

screenshots {
```

File: android/sample/src/androidTest/java/br/com/zup/beagle/sample/cucumber/robots/ScreenRobot.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    fun checkViewContainsText(text: String?, waitForText: Boolean = false): ScreenRobot {
+        if (waitForText){
+            WaitHelper.waitForWithElement(onView(withText(text)))
+        }
+
+        onView(Matchers.allOf(withText(text))).check(matches(isDisplayed()))
+
+        return this
+
+    }
```
Comment (diogooliveira-zup) [inline android/sample/src/androidTest/java/br/com/zup/beagle/sample/cucumber/robots/ScreenRobot.kt:38]: same here
```suggestion

        return this
    }
```

File: android/sample/src/androidTest/java/br/com/zup/beagle/sample/utils/TestUtils.kt [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        fun startBeagleActivity(activityTestRule: ActivityTestRule<SampleServerDrivenActivity>, url: String) {
+            val intent = activityTestRule.activity.newServerDrivenIntent<SampleServerDrivenActivity>(ScreenRequest(url))
+            activityTestRule.launchActivity(intent)
+        }
+    }
+
+
+}
```
Comment (diogooliveira-zup) [inline android/sample/src/androidTest/java/br/com/zup/beagle/sample/utils/TestUtils.kt:43]: same here
```suggestion
    }
}
```

File: android/sample/src/main/java/br/com/zup/beagle/sample/activities/CucumberActivity.kt [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            val intent = this.newServerDrivenIntent<SampleServerDrivenActivity>(ScreenRequest(url))
+            startActivity(intent)
+            finish()
+        }, 1500)
+        fragment_content.loadView(this, ScreenRequest(url))
+
+    }
+
+
+    fun bffUrl() = intent.extras?.getString(BFF_URL_KEY) ?: "http://10.0.2.2:8080/button"
```
Comment (diogooliveira-zup) [inline android/sample/src/main/java/br/com/zup/beagle/sample/activities/CucumberActivity.kt:49]: same here
```suggestion
    }

    fun bffUrl() = intent.extras?.getString(BFF_URL_KEY) ?: "http://10.0.2.2:8080/button"
```

**Review by diogooliveira-zup — CHANGES_REQUESTED** [review]

**Comment by viniciusguardieirozup** [comment]
This tests was added on 1.3.0-support

**Pull request closed without merging**